CMakeLists.txt

```diff
@@ -53,8 +67,12 @@ set (CMAKE_THREAD_PREFER_PTHREAD 1)
 
 if (WITH_GFLAGS)
   find_package (gflags 2.2.0)
-
   if (gflags_FOUND)
+    if (TARGET gflags::gflags)
+      set(GFLAGS_TARGET gflags::gflags)
+    elseif(TARGET gflags)
+      set(GFLAGS_TARGET gflags)
```

Comment:
Just curious, what is this logic doing? Why is the target sometimes `gflags::gflags` and other times just `gflags`, is this a side effect of version differences for the `glags` package maybe? I'm not too familiar with it.

Reply:
I guess, this is a workaround for gflags not providing an alias target when used as a subproject. 

@Mizux  Wouldn't it make sense to let the gflags folks add this alias instead of making this work using ugly hacks?

Reply:
Hi,
- Curently gflags-[target|config].cmake provide `gflags` imported target.
- There is no official findgflags.cmake module AFAIK
- I'm currently sending a [PR247 in gflags.git](https://github.com/gflags/gflags/pull/247) to use gflags::gflags instead (and providing alias as well when using as subproject)
- [gRPC](https://github.com/grpc/grpc/blob/master/cmake/gflags.cmake#L30) seems to already to look for gflags::gflags
- We have glog::glog protobuf::libprotobuf gRPC::grpc ...
- CMake preconise the use of namespace 
```
When providing imported targets, these should be namespaced (hence the Foo:: prefix); CMake will recognize that values passed to target_link_libraries() that contain :: in their name are supposed to be imported targets (rather than just library names), and will produce appropriate diagnostic messages if that target does not exist (see policy CMP0028).
```
src: https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html?highlight=namespace#a-sample-find-module (a ~bottom search for namespaced)

So all in all we have `gflags` currently but hope to have `gflags::gflags` soon -> here it is a workaround to be able to work during the transition since `find_package` don't provide a way to deal with this otherwise...

Reply:
I would prefer not adding the alias hack and wait until the `gflags::gflags` alias is added. No need for temporary workarounds.

Reply:
LGTM remove the commit for this PR